src/rust/ide/src/controller/project.rs

```diff
@@ -92,15 +94,31 @@ impl ExecutionContextsRegistry {
 
 
 
+// ===================
+// === ProjectData ===
+// ===================
+
+/// A structure containing the project's unique ID and name.
+#[derive(Debug,Clone)]
+pub struct ProjectData {
+    id   : Uuid,
+    name : ProjectName,
+}
+
+
+
 // ==========================
 // === Project Controller ===
 // ==========================
 
 /// Project controller's state.
 #[allow(missing_docs)]
-#[derive(Clone,CloneRef,Debug)]
+#[derive(Clone,CloneRef,Derivative)]
+#[derivative(Debug)]
 pub struct Handle {
-    pub project_name        : Rc<String>,
+    pub project_data        : Rc<RefCell<ProjectData>>,
```

Comment:
We can't change project id after opening, can we? If not, maybe it will be sufficient to keep only Rc<RefCell<ProjectName>>, and id as a separate immutable field?